### PR TITLE
Updated numerous project dependencies and parts

### PR DESCRIPTION
**Pull request by ChristopherCapito**
Hi,

first of all: great baseline work! I forked this to have a base for my embedded apps.

I noticed a few missing parts to make this usable as a general base.

1. Dependencies
- The project had a lot of duplicate and outdated dependencies. I cleaned things up and bumped all dependencies to the latest version, excluding React.
- All Shopify packages are now up to date, including Polaris
- Apollo packages have been removed in favor of the new @apollo packages.

I replaced all mentions of other apollo packages with the new Apollo package, and fixed all loose ends regarding the ApolloClient

I added husky for precommit linting of the project. 

Speaking of linting, using the next recommended base I fixed all open errors and warnings.

I also added a dockerfile and a docker compose file. This enables hot-reload development inside the container. No need to use vercel or a redis service provider, as this container setup includes redis.

TLDR: Now the project is usable out of the box.

Open issue: Session tokens do not get refreshed (like in the open issues)

### CODE REVIEW

**Comment by redochka** [comment]
Thank you so much for the great work, merged now to temporary mergingbranch because of different changes in different places, then I will integrate each change separately into the main branch.
Cheers

**Comment by redochka** [comment]
Properly merged to main just now :rocket: :tada: 
image

**Comment by ChristopherCapito** [comment]
Cool :D That was my first contribution to a public project ever. **slightly proud**
